bootstrap/bootstrap.go

```diff
@@ -472,7 +472,8 @@ func addRepositoryHostToSSHKnownHosts(sh *shell.Shell, repository string) {
 // setUp is run before all the phases run. It's responsible for initializing the
 // bootstrap environment
 func (b *Bootstrap) setUp(ctx context.Context) error {
-	span, ctx := tracetools.StartSpanFromContext(ctx, "environment", b.Config.TracingBackend)
+	spanName := b.implementationSpecificSpanName("phase/environment", "environment")
```

Comment:
Would `.` rather than `/` match other names better?
e.g. `phase.environment` instead of `phase/environment`.

Reply:
i had this thought as well, but in the opposite direction - the root span is `$org/$pipeline/$step_key`, and hook span names are `hook.$scope.$name`, so maybe we should rename the hook spans to `hook/$scope/$name`, and have everything be slash-delimited. 

imo slashes (`phase/environment`) better convey "this is the phase and the phase is called environment", but that could be my rails brain coming through, and that logic breaks down in the hook span case - `hook/plugin/command` doesn't really convey "this is a hook and its scope is plugin and its scope is command", so it's all a bit fuzzy really.

i'm happy to defer to your judgement here - i don't really have a strong opinion either way.

Reply:
another concern i have in this vague realm is the usability/understandability of the word "phase" - it's used in the agent internals, but i'm not sure it's a piece of jargon that we've actually defined, and i don't think it's in the docs. optimistically, i think that it's relatively easy to grok - it's a chunk of the agent's execution cycle for a job - but my judgement is probably clouded here by my familiarity with the agent.

Reply:
i agree with all of these - i think sticking to a more semantic name is good for consistency, i'll update all of these.

the one that might be a little tricky is calling the phase span eg just `command` for the command phase, because then you run into the issue of what do you call the span covering actually running the command, which is a child of the `command` span and a sibling of `pre-command` and `post-command`. I think we probably have a bit of an out here in that we could probably just get a bit more specific with this span name - eg `command execution` for command `repo checkout` for checkout, `artifact upload` for artifact, so i think it should be fine.

I'll implement this tomorrow morning :)

Reply:
Ah I hadn't found that naming advice, it's good 👍🏼